QUARKUS-5654.md

```diff
@@ -0,0 +1,34 @@
+# QUARKUS-5654 Update MP REST Client to 4.0
+
+JIRA: https://issues.redhat.com/browse/QUARKUS-5654
+
+Upstream PRs:
+- https://github.com/quarkusio/quarkus/pull/43959
+
+## Scope of the testing
+- New overloaded `baseUri(String uri)` method so users don’t have to call URI.create()
+- New `header(String name, Object value)` method for adding headers to Client instances
+- Clarify specification CDI support (`@RestClient` qualifier is not optional)
+- Add multi-part processing example to specification.
+- Clarify the specification that when a client injected as a CDI bean falls out of scope, the client is closed
+
+## Getting familiar with the feature
+Getting familiar with
+- https://download.eclipse.org/microprofile/microprofile-rest-client-4.0/microprofile-rest-client-spec-4.0.html
+
+## Existing test coverage
+* New overloaded `baseUri(String uri)` method so users don’t have to call URI.create() is covered upstream.
+* New `header(String name, Object value)` method for adding headers to Client instances is covered upstream.
+
+### Impact on test suites and testing automation
+None. Both new features are already covered in upstream tests. Examples and clarified tests doesn't affect Quarkus (and CDI support is handled this way in Quarkus since 2.13 at least).
```

Comment:
I think nor we or upstream have a native mode test for `RestClientBuilder#header` like is tested upstream in JVM https://github.com/quarkusio/quarkus/pull/43959/files#diff-eec444d1d58f2725293199b33aa951e7efca638abd66777b59fd5e40c1fda74bR44 . I don't expect issues. I basically say that `features are already covered in upstream tests` is true to some extent.

Reply:
Aren't upstream tests run in native?

Reply:
nothing in `extensions/..` is run in native